src/main/java/com/google/edith/interfaces/SearchService.java

```diff
@@ -0,0 +1,54 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.edith.interfaces;
+
+import com.google.appengine.api.datastore.Entity;
+import com.google.common.collect.ImmutableList;
+import com.google.edith.servlets.Item;
+import com.google.edith.servlets.Receipt;
+
+/** Operations for searching entities in Datastore with given filters. * */
+public interface SearchService {
+  /**
+   * Creates an immutable list of Receipt objects from entites of kind Receipt found in the
```

Comment:
nit: this javadoc is a little verbose. Hb ".. list of Receipt objects from receipt entities"

Reply:
Thanks for the suggestion. Changes made.